tinker/configuration.py

```diff
@@ -64,6 +77,55 @@ def iterate_generator(iterates: List[ConfigEntry]) -> Iterator[ConfigEntry]:
     return itertools.chain(*(config_generator(i) for i in iterates))
 
 
+def is_smqtk(value: Any) -> bool:
+    """Return `True` if `value` is a "smqtk" directive."""
+    return type(value) is dict and "smqtk" in value
+
+
+def smqtk_generator(smqtk_def: Dict[str, ConfigEntry]) -> Iterator[ConfigEntry]:
+    r"""
+    Find implementation of and instantiate a SMQTK class.
+
+    Instantiated with a config all values specified
+    by "smqtk" directive.
+
+
+    """
+
+    # get available implementations
+    # (from environment variable and entrypoint extension)
+    candidate_types = Pluggable.get_impls()
+
+    # check if our class matches any of the implementations
+    class_name = smqtk_def["class"]
+    matched_class = None  # TODO: can add typing here for matched_class
+    for candidate in candidate_types:
+        if candidate.__name__ == class_name:
+            matched_class = candidate
+            # TODO: add some sort of error checking if multiple classes
```

Comment:
`smqtk_core.configuration` "addresses" this by requiring its configuration scheme "type" field to be a fully qualified python path (module + type-name), at least for now. This TODO notionally exists there as well, if not already explicitly.

Reply:
How can we make use of that functionality here?

Reply:
There is a helper function in `smqtk_core.configuration`, `_type_to_key`, that is used internally to generate the key to a class type a more verbose, but fully qualified, `f"t.__module__}.{t.__name__}"` (replacing `t` with `candidate` in this context).

I'm not sure it is appropriate to utilize that helper here, but maybe using the fully qualified "rule" can address leaf-type name collisions. If more use of `make_default_config` and `from_config_dict` are utilized, then this functionality would be replaced by that.

Reply:
Thanks @Purg. I'll resolve this thread for now, and plan to open an issue when this difficulty begins to bite us more and more.